homeassistant/components/sensor/recorder.py

```diff
@@ -82,7 +80,9 @@ def _get_sensor_states(hass: HomeAssistant) -> list[State]:
     for state in all_sensors:
         if not is_entity_recorded(hass, state.entity_id):
             continue
-        if (state.attributes.get(ATTR_STATE_CLASS)) not in STATE_CLASSES:
+        try:
+            SensorStateClass(state.attributes.get(ATTR_STATE_CLASS))  # type: ignore[arg-type]
+        except ValueError:
```

Comment:
#87082 has been approved
```suggestion
        if not try_parse_enum(SensorStateClass, state.attributes.get(ATTR_STATE_CLASS)):
```